diff --git a/android/CMakeLists.txt b/android/CMakeLists.txt
index e2a5bfd..161695f 100644
--- a/android/CMakeLists.txt
+++ b/android/CMakeLists.txt
@@ -37,12 +37,14 @@ set(NATIVEVIEW_DIR "${CMAKE_SOURCE_DIR}/../Common/NativeView")
 set(PACKAGE_ROOT_DIR "${CMAKE_SOURCE_DIR}/..")
 set(CODEGEN_OUTPUT_DIR "${BUILD_DIR}/generated/source/codegen/jni")
 
+file(COPY "${COMMON_CPP_DIR}/reanimated" DESTINATION "${CMAKE_CURRENT_BINARY_DIR}/common_src")
+file(COPY "${NATIVEVIEW_DIR}" DESTINATION "${CMAKE_CURRENT_BINARY_DIR}/common_src")
 file(GLOB_RECURSE REANIMATED_COMMON_CPP_SOURCES CONFIGURE_DEPENDS
-     "${COMMON_CPP_DIR}/reanimated/*.cpp")
+     "${CMAKE_CURRENT_BINARY_DIR}/common_src/reanimated/*.cpp")
 file(GLOB_RECURSE REANIMATED_ANDROID_CPP_SOURCES CONFIGURE_DEPENDS
      "${ANDROID_CPP_DIR}/reanimated/*.cpp")
 file(GLOB_RECURSE REANIMATED_NATIVEVIEW_CPP_SOURCES CONFIGURE_DEPENDS
-     "${NATIVEVIEW_DIR}/react/renderer/components/rnreanimated/*.cpp")
+     "${CMAKE_CURRENT_BINARY_DIR}/common_src/NativeView/react/renderer/components/rnreanimated/*.cpp")
 
 find_package(fbjni REQUIRED CONFIG)
 find_package(ReactAndroid REQUIRED CONFIG)
@@ -68,6 +70,8 @@ target_compile_reactnative_options(reanimated PUBLIC)
 target_include_directories(
   reanimated
   PRIVATE "${COMMON_CPP_DIR}"
+          "${CMAKE_CURRENT_BINARY_DIR}/common_src"
+          "${CMAKE_CURRENT_BINARY_DIR}/common_src/NativeView"
           "${ANDROID_CPP_DIR}"
           "${PACKAGE_ROOT_DIR}"
           "${CODEGEN_OUTPUT_DIR}"
